plugin/src/main/cpp/extensions/openxr_fb_composition_layer_depth_test_extension_wrapper.cpp

```diff
@@ -0,0 +1,126 @@
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "extensions/openxr_fb_composition_layer_depth_test_extension_wrapper.h"
+
+#include <godot_cpp/classes/open_xrapi_extension.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+static const char *ENABLE_PROPERTY_NAME = "XR_FB_composition_layer_depth_test/enable";
+
+OpenXRFbCompositionLayerDepthTestExtensionWrapper *OpenXRFbCompositionLayerDepthTestExtensionWrapper::singleton = nullptr;
+
+OpenXRFbCompositionLayerDepthTestExtensionWrapper *OpenXRFbCompositionLayerDepthTestExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbCompositionLayerDepthTestExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbCompositionLayerDepthTestExtensionWrapper::OpenXRFbCompositionLayerDepthTestExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbCompositionLayerDepthTestExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_COMPOSITION_LAYER_DEPTH_TEST_EXTENSION_NAME] = &fb_composition_layer_depth_test_ext;
+	singleton = this;
+}
+
+OpenXRFbCompositionLayerDepthTestExtensionWrapper::~OpenXRFbCompositionLayerDepthTestExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbCompositionLayerDepthTestExtensionWrapper::_bind_methods() {
+}
+
+void OpenXRFbCompositionLayerDepthTestExtensionWrapper::cleanup() {
+	fb_composition_layer_depth_test_ext = false;
+}
+
+Dictionary OpenXRFbCompositionLayerDepthTestExtensionWrapper::_get_requested_extensions() {
+	Dictionary result;
+	for (auto ext : request_extensions) {
+		uint64_t value = reinterpret_cast<uint64_t>(ext.value);
+		result[ext.key] = (Variant)value;
+	}
+	return result;
+}
+
+uint64_t OpenXRFbCompositionLayerDepthTestExtensionWrapper::_set_viewport_composition_layer_and_get_next_pointer(const void *p_layer, const Dictionary &p_property_values, void *p_next_pointer) {
+	if (!fb_composition_layer_depth_test_ext) {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+
+	const XrCompositionLayerBaseHeader *layer = reinterpret_cast<const XrCompositionLayerBaseHeader *>(p_layer);
+
+	if (!layer_structs.has(layer)) {
+		layer_structs[layer] = {
+			XR_TYPE_COMPOSITION_LAYER_DEPTH_TEST_FB, // type
+			p_next_pointer, // next
+			0, // flags
+		};
+	}
+
+	XrCompositionLayerDepthTestFB *depth_test = layer_structs.getptr(layer);
+
+	if ((bool)p_property_values.get(ENABLE_PROPERTY_NAME, false)) {
+		depth_test->compareOp = XR_COMPARE_OP_LESS_FB; // Less depth = closer to the screen = keep this fragment
+		depth_test->depthMask = true;
+	}
```

Comment:
I don't think this will properly disable the depth test if the composition layer ever had it enabled. Since this never removes the struct from the next pointer chain, it'll still be there with the old settings, even if this code doesn't run

I think this either needs to (1) completely remove the struct from the next pointer chain when the option isn't enabled, or (2) update the struct to a "disabled state" if the option isn't enabled

I'd probably lean towards nr 1

Reply:
@dsnopek I've updated the logic to remove the struct when the option isn't enabled.